File: dfs/getImportance.java
/*
// Employee info
class Employee {
    // It's the unique id of each node;
    // unique id of this employee
    public int id;
    // the importance value of this employee
    public int importance;
    // the id of direct subordinates
    public List<Integer> subordinates;
};
*/
class Solution {
  public int getImportance(List<Employee> employees, int id) {
    for (Employee e: employees) {
      if (e.id == id) {
        if (e.subordinates.size() == 0) {
          return e.importance;
        }
        for (int subId: e.subordinates) {
          e.importance += getImportance(employees, subId);
        }
        return e.importance;
      }
    }
    return 0;
  }
}

File: daily/GetImportance.js
/**
 * Definition for Employee.
 * function Employee(id, importance, subordinates) {
 *     this.id = id;
 *     this.importance = importance;
 *     this.subordinates = subordinates;
 * }
 */

/**
 * @param {Employee[]} employees
 * @param {number} id
 * @return {number}
 */
var GetImportance = function (employees, id) {
  const map = new Map()
  for (const employee of employees) {
    const { id, importance, subordinates } = employee
    map.set(id, [importance, subordinates])
  }

  let ans = 0
  const queue = [id]
  while (queue.length) {
    const id = queue.shift()
    if (map.has(id)) {
      const [importance, subordinates] = map.get(id)
      ans += importance
      queue.push(...subordinates)
    }
  }
  return ans
}
